desktop/main/sources/fetchDiscovery.ts

```diff
@@ -23,18 +24,21 @@ import {
 import { handleIPC, handlerResult, makeSubscription } from '../rx.utils';
 import { downloadNodeConfig } from '../NodeConfig';
 
-export const fromNetworkConfig = (v) =>
-  from(downloadNodeConfig(v.conf)).pipe(
+export const fromNetworkConfig = (
+  net: Network,
+  $mainWindows: Observable<BrowserWindow>
+) =>
+  from(downloadNodeConfig(net.conf, $mainWindows)).pipe(
```

Comment:
Also, seems:
1. This function should be private (not used outside the module)
2. It should call `fetchNodeConfig` instead of `downloadNodeConfig`
3. Then `downloadNodeConfig` can have only one argument: URL (without optional `updateNodeConfig = false`)

As a consequence, we won't need to handle fs errors when we not dealing with them.
Fewer code changes, less (useless) coupling, and a more robust and clear workflow.